perllib/FixMyStreet/App/Controller/Admin.pm

```diff
@@ -1123,8 +1123,14 @@ sub user_edit : Path('user_edit') : Args(1) {
     my $user = $c->cobrand->users->find( { id => $id } );
     $c->detach( '/page_error_404_not_found' ) unless $user;
 
+    unless ( $c->user->is_superuser || ( $user->from_body && $c->user->has_permission_to('user_edit', $user->from_body->id) ) ) {
+        $c->detach('/page_error_403_access_denied', []);
+    }
+
     $c->stash->{user} = $user;
 
+    $c->stash->{available_permissions} = $c->cobrand->available_permissions;
```

Comment:
If this is only set if the user_manage_permissions permission exists, that simplifies the other checks perhaps.